sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/TokenValidatorInternal.cs

```diff
@@ -2,126 +2,77 @@
-                                foreach (AuthenticationEventJsonElement key in certsJson.GetPropertyValue<AuthenticationEventJsonElement>("keys").Elements)
-                                {
-                                    _cacheKeys.Add(new RsaSecurityKey(new RSAParameters { Modulus = FromBase64Url(key.GetPropertyValue("n")), Exponent = FromBase64Url(key.GetPropertyValue("e")) }));
-                                }
-                            }
-                        }
-                        finally
-                        {
-                            _semaphore.Release();
-                        }
-                        _lastSync = DateTime.Now;
-                    }
-                }).ConfigureAwait(false);
+                OpenIdConnectConfiguration oidcConfig =
+                    await OpenIdConnectConfigurationRetriever.GetAsync(
+                        address: oidcConfigAddress,
+                        cancel: CancellationToken.None).ConfigureAwait(false);
 
-                var result = handler.ValidateToken(accessToken, new TokenValidationParameters
+                if (oidcConfig == null)
                 {
-                    ValidateIssuer = true,
-                    ValidateAudience = true,
-                    ValidateIssuerSigningKey = true,
-                    ValidateLifetime = true,
-                    ValidIssuer = string.Format(CultureInfo.CurrentCulture, tokenSchemaVersion == SupportedTokenSchemaVersions.V2_0 ?
-                       serviceInfo.TokenIssuerV2 :
-                       serviceInfo.TokenIssuerV1, configurationManager.TenantId),
-                    ValidAudiences = authenticationAppIds,
-                    IssuerSigningKeys = _cacheKeys
-                }, out _);
+                    return (false, null);
+                }
 
-                //Here we validate the version and for version 1 validate that the appid claim matches the authorization party, if version 2, we validate that the azp claim matches the authorization party.
-                return (result.Claims.VerifyClaim("ver", $"{tokenSchemaVersion.GetDescription()}"),
-                    result.Claims.ToDictionary(x => x.Type, x => x.Value));
+                var handler = new JwtSecurityTokenHandler();
+
+                ClaimsPrincipal result = handler.ValidateToken(
+                    token: accessToken,
+                    validationParameters: new TokenValidationParameters
+                    {
+                        ValidateIssuerSigningKey = true,
+                        ValidIssuer = oidcConfig.Issuer,
+                        ValidAudience = configurationManager.AudienceAppId,
+                        IssuerSigningKeys = oidcConfig.SigningKeys
+                    },
+                    validatedToken: out _);
+
+                return (true, result.Claims.ToDictionary(x => x.Type, x => x.Value));
             }
             catch (Exception)
```

Comment:
so are we going to swallow if the validation fails? on line 70 we never check any status, so I assume it throws if validation doesn't succeed ?

Reply:
Yeap, If validation fails, it throws.

Reply:
So we swallow, why it failed, but not that it failed.

Reply:
Maybe we should add it as a log? thoughts... ?